src/libostree/ostree-fetcher.c

```diff
@@ -277,6 +277,24 @@ session_thread_config_flags (ThreadClosure *thread_closure,
 }
 
 static void
+on_authenticate (SoupSession *session, SoupMessage *msg, SoupAuth *auth,
+                 gboolean retrying, gpointer user_data)
+{
+  if (msg->status_code == SOUP_STATUS_PROXY_UNAUTHORIZED) {
+    SoupURI *uri = NULL;
+    g_object_get (session, SOUP_SESSION_PROXY_URI, &uri, NULL);
+    if (retrying)
+      {
+        g_autofree char *s = soup_uri_to_string (uri, FALSE);
+        g_warning ("Invalid username or password for proxy '%s'", s);
```

Comment:
I really don't like `g_warning()` as a general rule, particularly for things that can happen "normally" from things like incorrect passwords.

Do we really have no way to propagate this into a proper error back to the caller?  Maybe we have to do something like cache the error and return it later.